camera_app.py

```diff
@@ -94,21 +84,37 @@ def textToCommand(text):
 
 
 def textToSpeech(text):
-    # Turned into function to possibly use different, higher quality voice later
-    print(f"SPEAKING: {text}")
-    try:
-        engine = pyttsx3.init()
-        # Queue up the new text to be spoken
-        engine.say(text)
-        # Process the speech command and wait for it to finish
-        engine.runAndWait()
-    except RuntimeError:
-        # This can happen in rare cases, just ignore it to prevent a crash
-        pass
-    #pyttsx3.speak(text)  # Can be modified like https://pypi.org/project/pyttsx3/
+     #"""Uses a non-blocking PowerShell command to speak text without freezing the app."""
+     global last_speech_time
+     print(f"SAYING: {text}")
+     text = text.replace("'", "''")
+     command = f'powershell -Command "Add-Type –AssemblyName System.Speech; (New-Object System.Speech.Synthesis.SpeechSynthesizer).Speak(\'{text}\');"'
+    
+    # Popen runs the command without waiting, so the video doesn't freeze
+     subprocess.Popen(["powershell", "-Command", command], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+    
+    # Record the time we started speaking
+     last_speech_time = time.time()
 
-# Example of text to speech
-# textToSpeech("Testing the speech to text")
+def command_callback(recognizer, audio):
+    #"""This function is called in the background when speech is detected."""
+    global target_command, last_speech_time
+    
+    # If the app has spoken in the last 2 seconds, ignore any audio (it's an echo)
+    if time.time() - last_speech_time < 7:
```

Comment:
This hardcoded number does not match the commented number above... Please check this logic again or explain it to me